formats/gltf/writer.go

```diff
@@ -317,6 +325,112 @@ func rgbToFloatArr(c color.Color) [3]float64 {
 	}
 }
 
+func (w *Writer) AddScene(scene PolyformScene) error {
+	// Map to track unique meshes and their indices
+	meshIndices := make(map[*modeling.Mesh]int)
```

Comment:
I think it's valid to move this to the `Writer` struct. I think it's valid to have multiple scenes reference the same mesh

Reply:
And if it is moved to the struct, I think it'd be valid to move this behavior to the `AddMesh` function